### PR TITLE
New Sample: Display dimensions

### PR DESCRIPTION
**Pull request by shubham7109**
- PR to add a new sample `Display Dimensions` to Android samples.
- Sample [design](https://devtopia.esri.com/runtime/common-samples/tree/main/designs/DisplayDimensions)
- To test, run the sample and check if the toggle switches work as expected.

### CODE REVIEW

File: kotlin/display-dimensions/README.metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        "utility",
+        "DimensionLayer",
+        "MobileMapPackage"
+    ],
+    "language": "kotlin",
+    "provision_from": [
+        "https://www.arcgis.com/home/item.html?id=f5ff6f5556a945bca87ca513b8729a1e"
+    ],
+    "provision_to": [
+        "Edinburgh_Pylon_Dimensions.mmpk"
```
Comment (TADraeseke) [inline kotlin/display-dimensions/README.metadata.json:23]:
```suggestion
        "/Edinburgh_Pylon_Dimensions.mmpk"
```

File: kotlin/display-dimensions/README.metadata.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    ],
+    "relevant_apis": [
+        "DimensionLayer",
+        "MobileMapPackage"
+    ],
+    "snippets": [
+        "/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt"
```
Comment (TADraeseke) [inline kotlin/display-dimensions/README.metadata.json:30]:
```suggestion
        "src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt"
```

File: kotlin/display-dimensions/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+* DimensionLayer
+* MobileMapPackage
+
+## Offline data
+
+1. Download the data from [ArcGIS Online](https://arcgis.com/home/item.html?id=f5ff6f5556a945bca87ca513b8729a1e).
+2. Open your command prompt and navigate to the folder where you extracted the contents of the data from step 1.
+3. Execute the following commands:
+
+`adb push Edinburgh_Pylon_Dimensions.mmpk /Android/data/com.esri.arcgisruntime.sample.displaydimensions/cache/Edinburgh_Pylon_Dimensions.mmpk`
```
Comment (TADraeseke) [inline kotlin/display-dimensions/README.md:34]:
```suggestion
`adb push Edinburgh_Pylon_Dimensions.mmpk /Android/data/com.esri.arcgisruntime.sample.displaydimensions/files/Edinburgh_Pylon_Dimensions.mmpk`
```

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Thanks @shubham7109 -- Mostly just a few style suggestions, though it's definitely worth checking samples run in the sample viewer before opening the PR. The metadata had quite a few small issues that made it un-runnable.

Sample itself looks good though

File: kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        val mmpkFilePath =
+            File(externalCacheDir, getString(R.string.Edinburgh_Pylon_Dimensions)).path
+        val mobileMapPackage = MobileMapPackage(mmpkFilePath)
+
+        mobileMapPackage.addDoneLoadingListener {
+            // check the mmpk has loaded successfully and that it contains a map
+            if (mobileMapPackage.loadStatus == LoadStatus.LOADED && mobileMapPackage.maps.isNotEmpty()) {
+                // add the map from the mobile map package to the map view, and set a min scale to maintain dimension readability
+                mapView.map = mobileMapPackage.maps[0]
+                mapView.map.minScale = 35000.0
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt:69]: Increasing the min scale above `35000` does not display the dimensions of the dimension layer until zoomed into the scale of `35000` 
recording

**Comment by shubham7109** [comment]
Thanks for the suggestions @TADraeseke  , I will keep an eye out for the metadata changes in the future. I have gone ahead and added the suggestions.

File: kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.esri.arcgisruntime.layers.DimensionLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.MobileMapPackage
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.sample.displaydimensions.databinding.ActivityMainBinding
+import com.esri.arcgisruntime.sample.displaydimensions.databinding.DialogLayoutBinding
+import java.io.File
```
Comment (TADraeseke) [inline kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt:32]: this import's no longer needed

File: kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        mobileMapPackage.addDoneLoadingListener {
+            // check the mmpk has loaded successfully and that it contains a map
+            if (mobileMapPackage.loadStatus == LoadStatus.LOADED && mobileMapPackage.maps.isNotEmpty()) {
+                // add the map from the mobile map package to the map view, and set a min scale to maintain dimension readability
+                mapView.map = mobileMapPackage.maps[0]
+                mapView.map.minScale = 35000.0
+
+                // find the dimension layer within the map
+dimensionLayer =
+                    mapView.map.operationalLayers.firstOrNull { it is DimensionLayer } as DimensionLayer
```
Comment (TADraeseke) [inline kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt:73]: formatting issue here

File: kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+            } else {
+                val errorMessage = "Failed to load the mobile map package: " + mobileMapPackage.loadError.message
+                Log.e(TAG, errorMessage)
+                Toast.makeText(
+                    activityMainBinding.root.context,
+                    errorMessage,
+                    Toast.LENGTH_SHORT
+                ).show()
```
Comment (TADraeseke) [inline kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt:81]: Is the explicit context useful important? Could inline the whole line with `this`

Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show()

File: kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (TADraeseke) [reply kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt:69]: Ahh sorry--I conflated minScale and starting viewpoint scale here. When I ran the issue before this update the starting viewpoint was very zoomed in. It doesn't seem to be the case anymore--so nevermind this! Sorry!

**Review by TADraeseke — APPROVED** [review]
LGTM! Just a few small comments below. Cool sample, nice job!

File: kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt (hunk 6 above)
Comment (shubham7109) [reply kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt:73]: This is the result of Android Studio's file formatting. I shall revert it back to one line

**Pull request merged**